### ISSUE TITLE
Investigate support for Quarkus TLS Registry

### ISSUE THREAD

**Issue by jamesnetherton** [apache/camel-quarkus#6393]
Not sure if it's feasible since Camel already has `SSLContextParameters` etc. But might be worth investigating if it can be used to improve the general UX for TLS setup.

https://quarkus.io/guides/tls-registry-reference

**Comment by zhfeng** [comment]
I take a look at our extensions which depends on `quarkus-vertx-http` which has been added support for `quarkus-tls-registry`. There are some codes to check `sslEnabled`

https://github.com/apache/camel-quarkus/blob/f2e6f050e0ca72d5ae3c385e935caee64055aa9a/extensions/vertx-websocket/runtime/src/main/java/org/apache/camel/quarkus/component/vertx/websocket/VertxWebsocketRecorder.java#L58-L70

If it uses such `quarkus.tls.*` to config TLS, there is a potential issue? @jamesnetherton

**Comment by jamesnetherton** [comment]
> If it uses such `quarkus.tls.*` to config TLS, there is a potential issue? @jamesnetherton

Yes possibly if `quarkus.http.insecure-requests=disabled` since there'd be no plain HTTP for thw component to fall back on.

**Comment by jamesnetherton** [comment]
> Yes possibly if `quarkus.http.insecure-requests=disabled` since there'd be no plain HTTP for the component to fall back on.

Actually after a quick test, it can work with the sever secured by  `quarkus.tls.*`. What fails is the ability to do this check correctly:

https://github.com/apache/camel-quarkus/blob/f2e6f050e0ca72d5ae3c385e935caee64055aa9a/extensions/vertx-websocket/runtime/src/main/java/org/apache/camel/quarkus/component/vertx/websocket/VertxWebsocketRecorder.java#L89-L94

Not a huge issue because we [discourage](https://camel.apache.org/camel-quarkus/3.8.x/reference/extensions/vertx-websocket.html#extensions-vertx-websocket-usage-vert-x-websocket-consumers) explicit host / port configuration on the consumer. But I can see if I can improve things in this area.
